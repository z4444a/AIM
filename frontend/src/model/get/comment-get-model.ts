import BaseModel from '../base/base-model';
import NamedModel from '../base/named-model';

export interface CommentGetModel {
  datetime: Date;
  content: string;
  request: BaseModel;
  author: NamedModel;
}
