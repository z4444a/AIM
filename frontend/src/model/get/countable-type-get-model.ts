import NamedModel from '../base/named-model';

export default interface CountableTypeGetModel extends NamedModel {
  quantitative: boolean;
}
