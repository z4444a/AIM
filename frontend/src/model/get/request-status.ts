export enum RequestStatus {
  NEW = 1,
  IN_PROGRESS = 2,
  PROCESSED = 3,
  CANCELED = 4,
  PAUSED = 5,
}
