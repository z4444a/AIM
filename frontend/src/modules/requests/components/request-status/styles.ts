import { Theme, WithStyles } from '@material-ui/core';
import createStyles from '@material-ui/core/styles/createStyles';

export type Style = WithStyles<
  'container' | 'newReq' | 'inProgressReq' | 'canceledReq' | 'processedReq' | 'pausedReq'
>;

export const styles = (theme: Theme) =>
  createStyles({
    container: {
      display: 'flex',
      alignItems: 'center',
      width: '100px',
      '& > *:first-child': {
        marginRight: theme.spacing.unit,
      },
    },
    newReq: {
      color: '#6799d6',
    },
    inProgressReq: {
      color: '#6799d6',
    },
    canceledReq: {
      color: '#b3b3b3',
    },
    processedReq: {
      color: '#00ab33',
    },
    pausedReq: {
      color: '#CCCC00',
    },
  });
