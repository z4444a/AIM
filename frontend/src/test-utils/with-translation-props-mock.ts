import { WithTranslation } from 'react-i18next';

export function getMockWithTranslationProps(): WithTranslation {
  // eslint-disable-next-line
  const t = ((key: string) => `${key}_i18n`) as any;
  // eslint-disable-next-line
  const i18n = {} as any;
  const props: WithTranslation = {
    tReady: true,
    t,
    i18n,
  };

  return props;
}
