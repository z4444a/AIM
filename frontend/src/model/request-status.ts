import { RequestStatus } from './get/request-status';

export function RequestStatuses(t: (s: string) => string) {
  return [
    {
      key: RequestStatus.NEW.toString(),
      value: t('requestPage.status.new'),
    },
    {
      key: RequestStatus.IN_PROGRESS.toString(),
      value: t('requestPage.status.inWork'),
    },
    {
      key: RequestStatus.PROCESSED.toString(),
      value: t('requestPage.status.executed'),
    },
    {
      key: RequestStatus.CANCELED.toString(),
      value: t('requestPage.status.canceled'),
    },
    {
      key: RequestStatus.PAUSED.toString(),
      value: t('requestPage.status.paused'),
    },
  ];
}
