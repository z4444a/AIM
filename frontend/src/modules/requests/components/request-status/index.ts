import withStyles from '@material-ui/core/styles/withStyles';
import { RequestStatus } from './request-status';
import { styles } from './styles';
import { withTranslation } from 'react-i18next';

export default withStyles(styles)(withTranslation('common')(RequestStatus));
