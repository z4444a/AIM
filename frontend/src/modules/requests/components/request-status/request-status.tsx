import React, { ReactNode } from 'react';
import NamedModel from '../../../../model/base/named-model';
import { Style } from './styles';
import Timelapse from '@material-ui/icons/Timelapse';
import Brightness1 from '@material-ui/icons/Brightness1';
import Cancel from '@material-ui/icons/Cancel';
import { RequestStatus as ReqStatus } from '../../../../model/get/request-status';
import { WithTranslation } from 'react-i18next';

export interface Props {
  status: NamedModel;
}

export type InternalProps = Props & Style & WithTranslation;

export class RequestStatus extends React.PureComponent<InternalProps> {
  public render(): ReactNode {
    const { status } = this.props;
    switch (status.id) {
      case ReqStatus.NEW:
        return this.renderNewReq();
      case ReqStatus.IN_PROGRESS:
        return this.renderInProgressReq();
      case ReqStatus.PROCESSED:
        return this.renderProcessedReq();
      case ReqStatus.CANCELED:
        return this.renderCanceledReq();
      case ReqStatus.PAUSED:
        return this.renderPausedReq();
      default:
        return null;
    }
  }

  private renderContainer(renderInner: () => ReactNode) {
    const { classes } = this.props;
    return <div className={classes.container}>{renderInner()}</div>;
  }

  private renderNewReq(): ReactNode {
    const { classes, t } = this.props;
    return this.renderContainer(() => (
      <React.Fragment>
        <Brightness1 className={classes.newReq} />
        {t('requestPage.status.new')}
      </React.Fragment>
    ));
  }

  private renderInProgressReq(): ReactNode {
    const { classes, t } = this.props;
    return this.renderContainer(() => (
      <React.Fragment>
        <Timelapse className={classes.inProgressReq} />
        {t('requestPage.status.inWork')}
      </React.Fragment>
    ));
  }

  private renderProcessedReq(): ReactNode {
    const { classes, t } = this.props;
    return this.renderContainer(() => (
      <React.Fragment>
        <Brightness1 className={classes.processedReq} />
        {t('requestPage.status.executed')}
      </React.Fragment>
    ));
  }

  private renderCanceledReq(): ReactNode {
    const { classes, t } = this.props;
    return this.renderContainer(() => (
      <React.Fragment>
        <Cancel className={classes.canceledReq} />
        {t('requestPage.status.canceled')}
      </React.Fragment>
    ));
  }

  private renderPausedReq(): ReactNode {
    const { classes, t } = this.props;
    return this.renderContainer(() => (
      <React.Fragment>
        <Timelapse className={classes.pausedReq} />
        {t('requestPage.status.paused')}
      </React.Fragment>
    ));
  }
}
