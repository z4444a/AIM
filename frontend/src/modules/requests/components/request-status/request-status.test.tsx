import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import NamedModel from '../../../../model/base/named-model';
import { InternalProps, RequestStatus } from './request-status';
import { RequestStatus as ReqStatus } from '../../../../model/get/request-status';
import Timelapse from '@material-ui/icons/Timelapse';
import Brightness1 from '@material-ui/icons/Brightness1';
import Cancel from '@material-ui/icons/Cancel';
import { getMockWithTranslationProps } from '../../../../test-utils/with-translation-props-mock';

describe('RequestStatus', () => {
  function getComponentProps(status: NamedModel): InternalProps {
    const withTranslationProps = getMockWithTranslationProps();
    const props: InternalProps = {
      classes: {
        container: 'container',
        canceledReq: 'canceledReq',
        inProgressReq: 'inProgressReq',
        newReq: 'newReq',
        processedReq: 'processedReq',
      },
      status: status,
      ...withTranslationProps,
    };

    return props;
  }

  function getShallowWrapper(status: NamedModel) {
    const wrapper = shallow(<RequestStatus {...getComponentProps(status)} />);
    return wrapper;
  }

  it('should render correctly', () => {
    const wrapper = getShallowWrapper({ name: 'name', id: ReqStatus.NEW });
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should render NEW', () => {
    const wrapper = getShallowWrapper({ name: 'name', id: ReqStatus.NEW });
    expect(wrapper.find(Brightness1).exists()).toBe(true);
  });

  it('should render IN_PROGRESS', () => {
    const wrapper = getShallowWrapper({ name: 'name', id: ReqStatus.IN_PROGRESS });
    expect(wrapper.find(Timelapse).exists()).toBe(true);
  });

  it('should render CANCELED', () => {
    const wrapper = getShallowWrapper({ name: 'name', id: ReqStatus.CANCELED });
    expect(wrapper.find(Cancel).exists()).toBe(true);
  });

  it('should render PROCESSED', () => {
    const wrapper = getShallowWrapper({ name: 'name', id: ReqStatus.PROCESSED });
    expect(wrapper.find(Brightness1).exists()).toBe(true);
  });
});
