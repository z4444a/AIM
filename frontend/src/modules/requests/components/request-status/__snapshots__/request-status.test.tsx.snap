// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestStatus should render correctly 1`] = `
<div
  className="container"
>
  <pure(Brightness1Icon)
    className="newReq"
  />
  requestPage.status.new_i18n
</div>
`;
